drivers/dfi/driver.c

```diff
@@ -128,25 +128,37 @@ void dfi_ovr_traffic_cfg_reg_if(dfi_dev_t *dfi, dfi_ovr_traffic_cfg_t *cfg)
     reg_write(dfi->base + DDR_DFICH_CTRL1_M0_CFG__ADR + dfi->msr * DFI_MODE__OFFSET, cfg->val);
 }
 
-void dfi_set_init_complete_ovr_reg_if(bool enable)
+void dfi_set_init_complete_ovr_reg_if(bool override, uint8_t val)
 {
     uint32_t reg_val;
 
     reg_val = reg_read(WDDR_MEMORY_MAP_DFI + DDR_DFI_STATUS_IF_CFG__ADR);
-    reg_val = UPDATE_REG_FIELD(reg_val, DDR_DFI_STATUS_IF_CFG_SW_ACK_OVR, enable);
+    reg_val = UPDATE_REG_FIELD(reg_val, DDR_DFI_STATUS_IF_CFG_SW_ACK_VAL, val);
+    reg_write(WDDR_MEMORY_MAP_DFI + DDR_DFI_STATUS_IF_CFG__ADR, reg_val);
+    reg_val = UPDATE_REG_FIELD(reg_val, DDR_DFI_STATUS_IF_CFG_SW_ACK_OVR, override);
     reg_write(WDDR_MEMORY_MAP_DFI + DDR_DFI_STATUS_IF_CFG__ADR, reg_val);
 }
 
-void dfi_set_init_start_ovr_reg_if(bool enable)
+void dfi_set_init_start_ovr_reg_if(bool override, uint8_t val)
 {
     uint32_t reg_val;
 
     reg_val = reg_read(WDDR_MEMORY_MAP_DFI + DDR_DFI_STATUS_IF_CFG__ADR);
-    reg_val = UPDATE_REG_FIELD(reg_val, DDR_DFI_STATUS_IF_CFG_SW_REQ_OVR, enable);
+    reg_val = UPDATE_REG_FIELD(reg_val, DDR_DFI_STATUS_IF_CFG_SW_REQ_VAL, val);
+    reg_write(WDDR_MEMORY_MAP_DFI + DDR_DFI_STATUS_IF_CFG__ADR, reg_val);
+    reg_val = UPDATE_REG_FIELD(reg_val, DDR_DFI_STATUS_IF_CFG_SW_REQ_OVR, override);
     reg_write(WDDR_MEMORY_MAP_DFI + DDR_DFI_STATUS_IF_CFG__ADR, reg_val);
 }
 
 void dfi_get_init_start_status_reg_if(uint32_t *init_start)
 {
     *init_start = GET_REG_FIELD(reg_read(WDDR_MEMORY_MAP_DFI + DDR_DFI_STATUS_IF_STA__ADR), DDR_DFI_STATUS_IF_STA_REQ);
 }
+
+void dfi_ca_read_loopback_enable_reg_if(bool enable)
+{
+    uint32_t reg_val;
+    reg_val = reg_read(WDDR_MEMORY_MAP_DFI_CH0 +  DDR_DFICH_TOP_1_CFG__ADR);
+    reg_val = UPDATE_REG_FIELD(reg_val, DDR_DFICH_TOP_1_CFG_CA_RDDATA_EN, enable);
+    reg_write(WDDR_MEMORY_MAP_DFI_CH0 +  DDR_DFICH_TOP_1_CFG__ADR, reg_val);
+}
```

Comment:
This only supports CH0 correct?  Is loopback supported on CH1?

Reply:
I will add Channel 1.

Reply:
Actually, it's not supported. There's no DFI CH1 in the memory map. This is because of 1x32 configuration.